tracing/src/main/java/com/palantir/tracing/Tracer.java

```diff
@@ -154,6 +157,139 @@ public static void fastStartSpan(String operation) {
         fastStartSpan(operation, SpanType.LOCAL);
     }
 
+    /**
+     * Like {@link #startSpan(String, SpanType)}, but does not set or modify tracing thread state.
+     * This is an internal utility that should not be called directly outside of {@link DetachedSpan}.
+     */
+    static DetachedSpan detachInternal(String operation, SpanType type) {
+        Trace maybeCurrentTrace = currentTrace.get();
+        String traceId = maybeCurrentTrace != null
+                ? maybeCurrentTrace.getTraceId() : Tracers.randomId();
+        boolean sampled = maybeCurrentTrace != null
+                ? maybeCurrentTrace.isObservable() : sampler.sample();
+        return sampled
+                ? new SampledDetachedSpan(operation, type, traceId, getParentSpanId(maybeCurrentTrace))
+                : new UnsampledDetachedSpan(traceId);
+    }
+
+    private static Optional<String> getParentSpanId(@Nullable Trace trace) {
+        if (trace != null) {
+            Optional<OpenSpan> maybeOpenSpan = trace.top();
+            if (maybeOpenSpan.isPresent()) {
+                return Optional.of(maybeOpenSpan.get().getSpanId());
+            }
+        }
+        return Optional.empty();
+    }
+
+    private static final class SampledDetachedSpan implements DetachedSpan {
```

Comment:
Could we convert DetachedSpan into an abstract class, and inline Smapled/UnsampledDetachedSpan there? It would make the layout consistent with how Trace is currently implemented

Reply:
I could go either way on this. the DetachedSpan interface is meant to be relatively straightforward to read, and detached from implementation details because it's an API that will be consumed by developers using this library. Trace is an implementation detail, and isn't really consumed outside of this library.

If we moved these to the DetachedSpan class, we would also need to implement a package private accessor for the raw value of the Tracer trace thread-local, which could be dangerous if folks try to get clever.